src/main/java/seedu/multitasky/model/ModelManager.java

```diff
@@ -138,19 +138,27 @@ public synchronized void addEntry(ReadOnlyEntry entry)
     // @@author A0126623L
     @Override
     public void changeEntryState(ReadOnlyEntry entryToChange, Entry.State newState)
-            throws DuplicateEntryException, EntryNotFoundException {
-        _entryBook.changeEntryState(entryToChange, newState);
-        indicateEntryBookChanged();
+            throws DuplicateEntryException, EntryNotFoundException, OverlappingEventException {
+        try {
+            _entryBook.changeEntryState(entryToChange, newState);
+        } finally {
+            indicateEntryBookChanged();
```

Comment:
yeah, the `try` is because `changeEntryState` may throw an `OverlappingEventException`, but in that case everything should still go as per normal, only the user is alerted of overlapping entry. That means we still want `indicateEntryBookChanged()` to be executed. If `indicateEntryBookChanged()` is put inside the try block it won't be executed if an `OverlappingEventException` is thrown by `changeEntryState`.